ws-client/src/transport.rs

```diff
@@ -277,6 +268,31 @@ impl<'a> WsTransportClientBuilder<'a> {
 	}
 }
 
+impl<'a> TryFrom<WsConfig<'a>> for WsTransportClientBuilder<'a> {
+	type Error = WsHandshakeError;
+
+	fn try_from(config: WsConfig<'a>) -> Result<Self, Self::Error> {
+		let url =
+			url::Url::parse(&config.url).map_err(|e| WsHandshakeError::Url(format!("Invalid URL: {}", e).into()))?;
+		let mode = match url.scheme() {
+			"ws" => Mode::Plain,
+			"wss" => Mode::Tls,
+			_ => return Err(WsHandshakeError::Url("URL scheme not supported, expects 'ws' or 'wss'".into())),
+		};
+		let host = url.host_str().ok_or_else(|| WsHandshakeError::Url("No host in URL".into()))?.into();
```

Comment:
Isn't the error case taken care of by the `url::Url::parse()` call on line 276? Oh, I see, it is `None` for unix sockets... Ok, fine then.

The error is bit funny though, it's not really a handshaking error. Can we use a more appropriate error?

Reply:
> The error is bit funny though, it's not really a handshaking error. Can we use a more appropriate error?

indeed, let's fix that in another PR